plugins/inventory/opennebula.py

```diff
@@ -199,6 +199,9 @@ def _retrieve_servers(self, label_filter=None):
                     continue
 
             server['name'] = vm.NAME
+            server['id'] = vm.ID
+            if hasattr(vm.HISTORY_RECORDS, 'HISTORY'):
+                server['host'] = vm.HISTORY_RECORDS.HISTORY[-1].HOSTNAME
```

Comment:
What if there is no key `-1`? We should not rely on happy path. This can happen after OpenNebula database old records purge. I know there was bug in purge method, which deletes all records instead leaving last few.

Reply:
There was a check for it, I think it got removed after this comment: https://github.com/ansible-collections/community.general/pull/8532#discussion_r1782738993

IMO it's a good idea to re-add it, as there might be valid cases where there's no history (like the DB purge you mentioned).

Reply:
@feldsam Well in that case we can check for emptiness of a list adding additional check, something like this
what do you think?
```python
            if hasattr(vm.HISTORY_RECORDS, 'HISTORY') and vm.HISTORY_RECORDS.HISTORY:
                server['host'] = vm.HISTORY_RECORDS.HISTORY[-1].HOSTNAME
```